src/components/shared/Terminal.tsx

```diff
@@ -104,15 +113,131 @@ function Terminal(prop: {
     return executeTouch(path, true);
   }
 
+  function executeRm(path: string, flags: string, directory = false): string[] {
+    // Make a copy to avoid mutating the original (it's a state variable)
+    const fileSystemCopy = _.cloneDeep(prop.fileSystem);
+    const cwdCopy = fileSystemCopy.getFileSystemObjectFromPath(
+      prop.currentWorkingDirectory.path
+    );
+    let fileName = '';
+    let dir;
+
+    // If the path is absolute, find the directory to create the file in
+    // Otherwise, use the current working directory
+    if (path.includes('/')) {
+      const pathArr = path.split('/');
+      fileName = pathArr.pop() || '';
+      if (path.startsWith('/')) {
+        dir = fileSystemCopy.getFileSystemObjectFromPath(pathArr.join('/'));
+      } else {
+        dir = cwdCopy.getFileSystemObjectFromPath(pathArr.join('/'));
+      }
+    } else {
+      fileName = path;
+      dir = cwdCopy;
+    }
+
+    const file = dir?.getChild(fileName);
+    if (!file) {
+      return [`rm: '${path}': No such file or directory`];
+    }
+
+    if (
+      (directory && file.isEmpty()) ||
```

Comment:
Could we make each of these cases into a constant with the name explaining what it does? Kind of confused here lol

Reply:
Lol, yeah this part is kinda disturbing. I'll make a change to make it more clear what is happening here.